### ISSUE TITLE
Replace `notion-sdk-jvm` with custom Spring Rest Client

### ISSUE THREAD

**Issue by scordio** [scordio/spring-batch-notion#37]
Currently, [Notion SDK JVM](https://github.com/seratch/notion-sdk-jvm) is used as an underlying dependency to use the Notion API.

Although it gives the benefit of accessing the API from Java without additional effort, it has a few downsides:
* Brings Gson and Kotlin transitively
* Currently blocks the introduction of a module descriptor (#14)
* Could slow down the adoption of new API types released by Notion

An alternative could be to adopt one of the [Rest Client options](https://docs.spring.io/spring-framework/reference/integration/rest-clients.html) that Spring provides.

**Comment by sakshipatil1305** [comment]
is anyone working on this ?

**Comment by scordio** [comment]
Hi @sakshipatil1305, not yet!

Before starting here, a few things are ongoing, including the move to the spring-batch-extensions repo (see #53 and spring-projects/spring-batch-extensions#147) which will lead to archiving this repository.

Plus, I have the feeling the integration tests should be improved before such a refactoring can be done, as more complex use cases are currently delegated to the dependency I'd like to remove.

Let me ping you once this topic is ready to be picked up, if you don't mind.

**Comment by scordio** [comment]
Superseded by spring-projects/spring-batch-extensions#154.
